8-algoruthm-and-data-structures-2/hometask/src/JobRunner.ts

```diff
@@ -0,0 +1,79 @@
+interface Job {
+    priority: number;
+}
+
+class JobRunner {
+    private jobs: Job[];
+    
+    constructor(jobs: Job[]) {
+        this.jobs = jobs;
+    }
+
+    run(): void {
+        const sortedJobs = this.mergeSortJobByPriority(this.jobs);
+        this.doJobs(sortedJobs);
+    }
+
+    doJobs (jobs: Job[]){
```

Comment:
I suppose to run the job you have to add that job first. So in this implementation you have JobRunner which runs jobs passed before and at the same time doJobs method to run jobs from separated array (which sounds strange).
I'd make JobRunner class with methods to register jobs, run jobs (or run specific job by providing its id) or delete jobs from internal queue (but client shouldn't see details of internal implementation anyway).